gulp.nginclude.js

```diff
@@ -0,0 +1,86 @@
+/* jshint node:true */
+'use strict';
+
+var fs = require('fs');
+var path = require('path');
+var cheerio = require('cheerio');
+var glob = require('glob');
+var gutil = require('gulp-util');
+var through = require('through2');
+//var extend = require('lodash.assign');
+var extend = require('util')._extend;
+var PluginError = gutil.PluginError;
+
+module.exports = function(options) {
+    options = extend({trim: true}, options);
+
+    function transform(file, encoding, next) {
+        var cwd = process.cwd();
+
+        if (file.isNull()) {
+            return next(null, file); // pass along
+        }
+
+        var $ = cheerio.load(file.contents.toString('utf8'), {decodeEntities: false});
+
+        // This function receives an ng-include src and tries to read it from the filesystem
+        function readSource(src) {
+            if (options.assetsDirs && options.assetsDirs.length) {
+                var srcTheme = path.join(options.assetsDirs[0], src);
+                var match = glob.sync(srcTheme, {cwd: cwd});
+                if (!match.length) {
+                    srcTheme = path.join(options.assetsDirs[1], src);
+                    match = glob.sync(srcTheme, {cwd: cwd});
+                }
+            }
+
+            if (!match.length) return next(new PluginError('gulp-nginclude', 'File "' + srcTheme + '" not found'));
+            return fs.readFileSync(path.join(cwd, match[0])).toString();
+        }
+
+        function processTag(i, ng) {
+            var $ng = $(ng);
+            var src = $ng.attr('src') || $ng.attr('ng-include');
+
+            // Remove old ng-include attributes so Angular doesn't read them
+            $ng.removeAttr('src').removeAttr('ng-include');
+
+            var before = '\n<!-- ngInclude: ' + src + ' -->\n';
+            var after = '\n<!--/ngInclude: ' + src + ' -->\n';
+            var include = readSource(src);
+            if (options.trim && include) {
+                include = include.trim();
+            }
+            $ng.html(before + include + after);
+            return true;
+        }
+
+        // Use while to make the task recursive
```

Comment:
I think we should use another inline named function expression like above.  This way we follow the rule of ** do one thing **